lib/streamlit/components/v1/component_registry.py

```diff
@@ -62,13 +63,19 @@ def declare_component(
 
     Returns
     -------
-    CustomComponent
+    CustomComponent or None
         A CustomComponent that can be called like a function.
         Calling the component will create a new instance of the component
         in the Streamlit app.
+        Returns None if no ScriptRunContext exists; this can happen when CustomComponents are executed as standalone commands, e.g. for testing.
 
     """
 
+    # the ctx can be None if a custom component script is run outside of Streamlit, e.g. via 'python ...'
+    ctx = get_script_run_ctx()
```

Comment:
Do we actually have to return `None` here or could we just run everything but just not register the component? e.g.:

```python
ctx = get_script_run_ctx()
if ctx:
    get_instance().component_registry.register_component(component)
```

Reply:
That's a good point and should work 👍 I will check it out

Reply:
Updated it!